go.mod

```diff
@@ -6,17 +6,17 @@ replace github.com/ipfs/go-verifcid => github.com/celestiaorg/go-verifcid v0.0.1
 
 require (
 	github.com/BurntSushi/toml v1.1.0
-	github.com/celestiaorg/celestia-app v0.4.0
+	github.com/celestiaorg/celestia-app v0.5.0-rc-2
 	github.com/celestiaorg/go-libp2p-messenger v0.1.0
 	github.com/celestiaorg/nmt v0.8.0
-	github.com/celestiaorg/rsmt2d v0.4.0
-	github.com/cosmos/cosmos-sdk v0.44.3
+	github.com/celestiaorg/rsmt2d v0.5.0
+	github.com/cosmos/cosmos-sdk v0.46.0-beta2.0.20220418184507-c53157dd63f6
 	github.com/dgraph-io/badger/v2 v2.2007.4
 	github.com/gammazero/workerpool v1.1.2
 	github.com/gogo/protobuf v1.3.3
 	github.com/gorilla/mux v1.8.0
 	github.com/hashicorp/go-retryablehttp v0.7.1-0.20211018174820-ff6d014e72d9
-	github.com/hashicorp/golang-lru v0.5.4
+	github.com/hashicorp/golang-lru v0.5.5-0.20210104140557-80c98217689d
```

Comment:
If this isn't an indirect dependency, it shouldn't use a non-tag.

Reply:
True. Still, this is automatically done after we execute 
`go mod tidy -compat=1.17`
otherwise, using `go mod tidy` will cause such messages:
```
To upgrade to the versions selected by go 1.16, leaving some packages unresolved:
	go mod tidy -e -go=1.16 && go mod tidy -e -go=1.17
If reproducibility with go 1.16 is not needed:
	go mod tidy -compat=1.17
```

I still don't understand how go is deciding on versioning names as the diff between 0.5.4 and this(0.5.5-xxxx) is just 2 commits without any go.mod changes
1. eb529947af531eb529020ba979a7a887338904d1
2. 80c98217689d6df152309d574ccc682b21dc802c

Reply:
Where is `golang-lru` actually used/imported?

Reply:
> Where is `golang-lru` actually used/imported?

Somewhere in core 🫤

Reply:
No, we use it in header.Store

Reply:
Still unclear to me why `go mod tidy -compat=1.17` bumps this to an exact commit instead of just using the latest tag.

Reply:
This is not true anymore. Try to checkout locally. This is something new to me and I didn't have enough time to dive and understand what it the reasoning behind this thing
> It is not golang that decides here but the developer that imports this.

Reply:
Really, if you don't add the commit at the end. The go does not compile and throws an error above

Reply:
That is probably due to some indirect dep conflict then. Internally sth uses that newer version and there is no way to resolve this other than using the same version. Resolving this thread for now!